src/storage/txn/flow_controller/tablet_flow_controller.rs

```diff
@@ -202,25 +198,17 @@ impl FlowInfoDispatcher {
                                 }
                             };
                         }
-                        Ok(FlowInfo::Destroyed(region_id, suffix)) => {
-                            let mut remove_limiter = false;
-                            {
-                                let mut checkers = flow_checkers.as_ref().write().unwrap();
-                                if let Some(checker) = checkers.get_mut(&region_id) {
-                                    if checker.tablet_suffix() == suffix {
-                                        checkers.remove(&region_id);
-                                        remove_limiter = true;
-                                    }
-                                }
-                            }
-                            if remove_limiter {
-                                limiters.as_ref().write().unwrap().remove(&region_id);
-                            }
+                        Ok(FlowInfo::Destroyed(region_id)) => {
+                            let mut checkers = flow_checkers.as_ref().write().unwrap();
+                            checkers.remove(&region_id);
+                            limiters.as_ref().write().unwrap().remove(&region_id);
```

Comment:
Release checkers before locking limiters, or there could be deadlock if not careful.